spec/requests/companies_spec.rb

```diff
@@ -4,9 +4,19 @@
 
 RSpec.describe 'Companies', type: :request do
   describe 'GET /index' do
-    it 'returns http success' do
+    it 'render template' do
       get '/companies'
+
       expect(response).to have_http_status(:success)
+      expect(response).to render_template(:index)
+      expect(response.body).to include('Companies')
+      expect(response.body).to include('Joint Stock Company')
+    end
+
+    it 'does not render a different template' do
+      get '/companies'
+
+      expect(response).to_not render_template(:show)
```

Comment:
In my opinion it is not needed. This is "requests" spec, so you are testing the endpoint more from the user perspective (even the router is under tests), and the name of the template is internal implementation's detail that is not important from this test's perspective.

```suggestion
```

Reply:
Ok, so I don't need this : `expect(response).to render_template(:index)` as well.
Then don't need  `gem 'rails-controller-testing'`